jujugui/static/gui/src/app/init/component-renderers-mixin.js

```diff
@@ -0,0 +1,1080 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const yui = window.yui;
+
+const ComponentRenderersMixin = (superclass) => class extends superclass {
```

Comment:
Also, please mention the set of attributes and methods expected to be present in the provided superclass. This way we establish a contract that must be respected by the given superclass. For instance, in this case the superclass must have the db and topology attributes, etc. To be honest, I like self-sufficient mixins more, like, mixins for adding a specific functionality rather than mixins for putting some kinda related methods together in a separate file. But if the mixin relies on something existing in the provided superclass then we must absolutely document it in the factory.